modules/apps/layout/layout-service/src/main/java/com/liferay/layout/internal/upgrade/remover/LayoutDuplicateRemover.java

```diff
@@ -0,0 +1,52 @@
+package com.liferay.layout.internal.upgrade.remover;
+
+import com.liferay.portal.db.remover.DuplicateRemover;
+import com.liferay.portal.db.remover.PortalDuplicateRemover;
+import com.liferay.portal.kernel.dao.orm.Query;
+import com.liferay.portal.kernel.dao.orm.Session;
+import com.liferay.portal.kernel.model.Layout;
+import com.liferay.portal.kernel.service.persistence.LayoutPersistence;
+import com.liferay.portal.kernel.util.StringBundler;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component(property = "service.tables=portal", service = DuplicateRemover.class)
```

Comment:
The property "service.tables" will be set to the tableName that corresponds for each indexSQLQuery,  that way the tableName can control which component is used.

Reply:
This a good idea to get the right service when creating the unique index